File: Calculator.java
/*
=====================================================
PRACTICAL: Simple Calculator using Java
=====================================================

AIM:
- To create a calculator program in Java that performs
  basic arithmetic operations using user input.

OBJECTIVE:
- To use switch-case for menu-driven calculator.
- To perform addition, subtraction, multiplication, division.
- To take input from user at runtime.

TOOLS:
- JDK 8+ (javac, java)
- Any IDE or text editor

ALGORITHM:
1) Import Scanner class for input.
2) Display menu for arithmetic operations.
3) Read user’s choice.
4) Take two numbers as input.
5) Use switch-case to perform selected operation.
6) Display the result.
7) Repeat until user exits.

STEPS TO RUN:
- Save as: Calculator.java
- Compile: javac Calculator.java
- Run:     java Calculator

=====================================================
*/

import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        boolean again = true;

        while (again) {
            System.out.println("\n=== Simple Calculator ===");
            System.out.println("1. Addition");
            System.out.println("2. Subtraction");
            System.out.println("3. Multiplication");
            System.out.println("4. Division");
            System.out.println("5. Exit");
            System.out.print("Enter your choice: ");
            int choice = sc.nextInt();

            if (choice == 5) {
                System.out.println("Exiting... Thank you!");
                break;
            }

            System.out.print("Enter first number: ");
            double num1 = sc.nextDouble();
            System.out.print("Enter second number: ");
            double num2 = sc.nextDouble();

            double result = 0;
            boolean valid = true;

            switch (choice) {
                case 1: result = num1 + num2; break;
                case 2: result = num1 - num2; break;
                case 3: result = num1 * num2; break;
                case 4: 
                    if (num2 != 0) {
                        result = num1 / num2;
                    } else {
                        System.out.println("Error: Division by zero not allowed.");
                        valid = false;
                    }
                    break;
                default:
                    System.out.println("Invalid choice!");
                    valid = false;
            }

            if (valid) {
                System.out.println("Result = " + result);
            }

            System.out.print("Do you want to continue? (y/n): ");
            String ans = sc.next();
            if (!ans.equalsIgnoreCase("y")) {
                again = false;
            }
        }

        sc.close();
    }
}

/*
================= SAMPLE OUTPUT =================

=== Simple Calculator ===
1. Addition
2. Subtraction
3. Multiplication
4. Division
5. Exit
Enter your choice: 1
Enter first number: 10
Enter second number: 20
Result = 30.0
Do you want to continue? (y/n): y

=== Simple Calculator ===
1. Addition
2. Subtraction
3. Multiplication
4. Division
5. Exit
Enter your choice: 4
Enter first number: 50
Enter second number: 5
Result = 10.0
Do you want to continue? (y/n): n

Exiting... Thank you!

=================================================

RESULT:
- Successfully created a calculator in Java using switch-case.
- Program takes user input, performs arithmetic operations,
  and displays result.
=================================================
*/

File: python/calculator.py
# ======================================================
# 📘 Practical – 20 : Simple Calculator using Functions
# ======================================================

# 🔹 Aim / Objective:
#   To write a Python program for a simple calculator 
#   using functions to perform +, -, *, / operations.

# 🔹 Tools Required:
#   - Python 3.x Interpreter
#   - Text Editor / IDE (IDLE, VS Code, PyCharm etc.)

# 🔹 Algorithm:
#   1. Start program
#   2. Define functions for addition, subtraction, 
#      multiplication, and division
#   3. Take two numbers from user
#   4. Take operator choice from user
#   5. Call respective function
#   6. Display result
#   7. Stop program

# 🔹 Steps:
#   Step 1: Start program
#   Step 2: Define functions add(), sub(), mul(), div()
#   Step 3: Take two numbers from user
#   Step 4: Take operator input
#   Step 5: Call respective function
#   Step 6: Print result
#   Step 7: Stop program

# ======================================================
# 🔹 Code
# ======================================================

# Step 1: Define functions
def add(x, y):
    return x + y

def sub(x, y):
    return x - y

def mul(x, y):
    return x * y

def div(x, y):
    if y == 0:
        return "Error: Division by Zero"
    else:
        return x / y

# Step 2: Take input from user
a = float(input("Enter first number: "))
b = float(input("Enter second number: "))
op = input("Enter operator (+, -, *, /): ")

# Step 3: Perform operation
if op == '+':
    print("Result:", add(a, b))
elif op == '-':
    print("Result:", sub(a, b))
elif op == '*':
    print("Result:", mul(a, b))
elif op == '/':
    print("Result:", div(a, b))
else:
    print("Invalid Operator")

# ======================================================
# 🔹 Sample Output
# ======================================================
# Case 1:
# Enter first number: 10
# Enter second number: 5
# Enter operator (+, -, *, /): +
# Result: 15.0
#
# Case 2:
# Enter first number: 8
# Enter second number: 0
# Enter operator (+, -, *, /): /
# Result: Error: Division by Zero
# ======================================================

# 🔹 Result:
# The program successfully performs addition, subtraction, 
# multiplication, and division using functions.
# ======================================================
